src/Psl/Iter/chunk.php

```diff
@@ -26,27 +27,15 @@
  * @psalm-return Iterator<int, list<T>>
  *
  * @throws Psl\Exception\InvariantViolationException If $size is negative.
+ *
+ * @deprecated since 1.2, use Vec\chunk instead.
```

Comment:
Curious, why is the iterable version deprecated? When processing a large data stream, it be can be useful to have a lazy chunk fn that doesn't consume the entire data set into memory.